code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,458 @@
+
+/obj/item/nemesis_mine
+	name = "energy mine"
+	desc = "A high-tech trap, designed to electrocute any intruders that step on it."
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "nemesis_trap_inactive"
+	w_class = WEIGHT_CLASS_SMALL
+
+	var/active = FALSE
+	var/overloaded = FALSE
+	var/obj/item/clothing/glasses/hud/security/sunglasses/nemesis/glasses
+
+/obj/item/nemesis_mine/Crossed(mob/living/AM)
+	. = ..()
+
+	if(!istype(AM) || !isturf(loc) || !active)
+		return
+
+	if(AM.movement_type & FLYING)
+		return
+
+	trigger(AM)
+
+/obj/item/nemesis_mine/attack_self(mob/user)
+	if(!ishuman(user))
+		return ..()
+
+	var/mob/living/carbon/human/owner = user
+	var/obj/item/clothing/gloves/rapid/nemesis/gloves = owner.get_item_by_slot(ITEM_SLOT_GLOVES)
+	if(!gloves || !istype(gloves))
+		return ..()
+
+	if(overloaded)
+		gloves.gain_charge(round(NEMESIS_CHARGE_PER_MINE * 0.5)) //Small refund
```

Comment:
if this is supposed to be a refund, why is gloves.lose_charge(NEMESIS_CHARGE_PER_MINE) in the else statement